File: src/main/java/ro/capan/Recipes/domain/SideDish.java
package ro.capan.Recipes.domain;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.*;

@Entity
@Table(name = "side_dish")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class SideDish {
    @Id
    @GeneratedValue(strategy = GenerationType.SEQUENCE)
    private Long id;
    private String name;
    @Column(name = "has_meat")
    private Boolean hasMeat;
    private String notes;
}

File: recipe-randomizer/src/app/utils/SideDish.ts
export class SideDish {
  id: number;
  name: string;
  hasMeat: boolean;
  notes: string;

  constructor(id?: number, name?: string, hasMeat?: boolean, notes?: string) {
    this.id = id??0;
    this.name = name??'';
    this.hasMeat = hasMeat??true;
    this.notes = notes??'';
  }
}
